src/input/parse_json.rs

```diff
@@ -146,7 +146,7 @@ impl<'de> Deserialize<'de> for JsonInput {
 
                                 if let JsonInput::String(s) = &first_value {
                                     // Normalize the string to either an int or float
-                                    let normalized = if s.contains('.') {
+                                    let normalized = if s.contains('.') || s.contains('e') {
```

Comment:
Maybe there's a better way to do this check, or maybe there's a way to better-infer from serde_json itself that the value should be a number..? @davidhewitt @samuelcolvin @adriangb

Reply:
Okay after some unnecessarily thorough investigation I'm pretty confident this approach is plenty fast